src/strings.cpp

```diff
@@ -903,733 +901,733 @@ static void FormatString(StringBuilder &builder, const char *str_arg, StringPara
+					break;
 				}
-				break;
-
-			case SCC_PLURAL_LIST: { // {P}
-				int plural_form = *str++;          // contains the plural form for this string
-				size_t offset = orig_offset + (byte)*str++;
-				int64_t v = args.GetParam(offset); // contains the number that determines plural
-				str = ParseStringChoice(str, DeterminePluralForm(v, plural_form), builder);
-				break;
-			}
 
-			case SCC_ARG_INDEX: { // Move argument pointer
-				args.SetOffset(orig_offset + (byte)*str++);
-				break;
-			}
+				case SCC_ARG_INDEX: { // Move argument pointer
+					args.SetOffset(orig_offset + (byte)*str++);
+					break;
+				}
 
-			case SCC_SET_CASE: { // {SET_CASE}
-				/* This is a pseudo command, it's outputted when someone does {STRING.ack}
-				 * The modifier is added to all subsequent GetStringWithArgs that accept the modifier. */
-				next_substr_case_index = (byte)*str++;
-				break;
-			}
+				case SCC_SET_CASE: { // {SET_CASE}
+					/* This is a pseudo command, it's outputted when someone does {STRING.ack}
+					 * The modifier is added to all subsequent GetStringWithArgs that accept the modifier. */
+					next_substr_case_index = (byte)*str++;
+					break;
+				}
 
-			case SCC_SWITCH_CASE: { // {Used to implement case switching}
-				/* <0x9E> <NUM CASES> <CASE1> <LEN1> <STRING1> <CASE2> <LEN2> <STRING2> <CASE3> <LEN3> <STRING3> <STRINGDEFAULT>
-				 * Each LEN is printed using 2 bytes in big endian order. */
-				uint num = (byte)*str++;
-				while (num) {
-					if ((byte)str[0] == case_index) {
-						/* Found the case, adjust str pointer and continue */
-						str += 3;
-						break;
+				case SCC_SWITCH_CASE: { // {Used to implement case switching}
+					/* <0x9E> <NUM CASES> <CASE1> <LEN1> <STRING1> <CASE2> <LEN2> <STRING2> <CASE3> <LEN3> <STRING3> <STRINGDEFAULT>
+					 * Each LEN is printed using 2 bytes in big endian order. */
+					uint num = (byte)*str++;
+					while (num) {
+						if ((byte)str[0] == case_index) {
+							/* Found the case, adjust str pointer and continue */
+							str += 3;
+							break;
+						}
+						/* Otherwise skip to the next case */
+						str += 3 + (str[1] << 8) + str[2];
+						num--;
 					}
-					/* Otherwise skip to the next case */
-					str += 3 + (str[1] << 8) + str[2];
-					num--;
+					break;
 				}
-				break;
-			}
 
-			case SCC_REVISION: // {REV}
-				builder += _openttd_revision;
-				break;
+				case SCC_REVISION: // {REV}
+					builder += _openttd_revision;
+					break;
 
-			case SCC_RAW_STRING_POINTER: { // {RAW_STRING}
-				const char *raw_string = args.GetNextParameterString();
-				/* raw_string can be(come) nullptr when the parameter is out of range and 0 is returned instead. */
-				if (raw_string == nullptr) {
-					builder += "(invalid RAW_STRING parameter)";
+				case SCC_RAW_STRING_POINTER: { // {RAW_STRING}
+					const char *raw_string = args.GetNextParameterString();
+					FormatString(builder, raw_string, args);
 					break;
 				}
```

Comment:
It could also be `nullptr` when you, for whatever reason, pass `nullptr` to `SetDParamStr`. You can currently do that, and it will also trigger this code.
The question is whether we want to game to crash here, or should we maybe add an `assert` in `SetDParamStr` where applicable?

Reply:
If it's still possible then I'll put it back to how it was I was misinformed :-)

Reply:
Maybe replace the comment with `/* raw_string can be nullptr. */`, i.e. remove the bit about the parameter being out of range.

Reply:
My bad, the comment was misleading :)